### DISCUSSION TITLE
[API Request] Explicit tag (for songs and maybe albums?)

### DISCUSSION THREAD

**Discussion by certuna** [opensubsonic/open-subsonic-api#70] [API requests]
### Type of change

API Clarification

### Proposal description

This is a feature that the APIs of most streaming services ([Tidal](https://apiref.developer.tidal.com/apiref?spec=catalogue&ref=get-track)/Spotify/Apple/etc) have and there seems to be demand for adding this to Navidrome https://github.com/navidrome/navidrome/issues/1357 . Would not be very complex to add on a per-song basis (just an optional boolean), question maybe if we want albums to have this attribute too? And if we would want clients to filter responses by it?

Note: in Apple Music-world this tag is called Content Rating (stored in the `rtng` MP4 tag field) or "iTunes Advisory", but since this is easily confused with the feature of users setting 5-star ratings, I think `explicit` is a better attribute name than `rating` or `contentRating` or `advisory`. But please tell me if I'm wrong.

### Backward compatibility

- [ ] No backward compatibility impact.

### API details

In `Child` [OS]:
`<xs:attribute name="explicit" type="xs:boolean" use="optional"/>`

In `search3` [OS], `getSimilarSongs`, `getSongsByGenre`, `getRandomSongs`, `getPlaylist`, `getAlbum`, `getStarred`, `getStarred2`:
optional parameter `explicit`

**Comment by sammy-ette** [comment]
An album would be explicit if at least 1 song in it is also explicit. I think having explicit on the album details makes sense.

**Comment by Tolriq** [comment]
Hum I missed that one, is there different criteria for songs for this? Like the PEGI / MPA stuff?
From a quick search it seems no, just https://en.wikipedia.org/wiki/Parental_Advisory ?

If there's only that then except the naming (explicitContent? because explicit by itself is not very clear) we can easily expose this to child and AlbumID3.

For the filter parameter this is another story as we need to build a new proper api for that.

**Reply by Tolriq** [reply]
So after adding support for this in Symfonium and reading a little more about how those works I propose something like:

explicitStatus: String -> "" for unknown, "Explicit" for explicit and "Clean" for clean.

Server side the tag: "ITUNESADVISORY" gives 1 = Explicit, 2 = Clean and for MP4 "rtng" 1 or 4 = Explicit, 2 = Clean.

"rtng" 1 is not in the specs but due to abuse and mix between the 2 tags by some users and software this covers most cases.

Please comment or +1 so I do the PR.
